.gitignore

```diff
@@ -1,8 +1,38 @@
 /nbproject/private/
 /build/
 /dist/
+/bin/
 *.class
+/.idea
+.classpath
+.project
+manifest.mf
+
+notepad.iml
 
 # Mac OS System
 .DS_Store*
 ._*
+
+# Compiled class file
+*.class
+
+# Log file
+*.log
+
+# BlueJ files
+*.ctxt
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
```

Comment:
Not sure the below file extensions are really necessary for this kind of project. Normally we won't have a .zip., .rar, etc.